es-app/src/guis/GuiCollectionSystemsOptions.cpp

```diff
@@ -198,27 +198,28 @@ void GuiCollectionSystemsOptions::applySettings()
 	bool prevBundle = Settings::getInstance()->getBool("UseCustomCollectionsSystem");
 	bool prevShow = Settings::getInstance()->getBool("CollectionShowSystemInfo");
 	bool outShow = toggleSystemNameInCollections->getState();
+
 	// need to check if collection is enabled
 	std::string enabledCollectionName = "";
-	if (defaultScreenSaverCollection->getSelectedObjects().size() != 0)
-	{
-		std::string selectedCollection = defaultScreenSaverCollection->getSelectedObjects().at(0);
-		if (selectedCollection != "") {
-			std::vector<std::string> enabledCollections = customOptionList->getSelectedObjects();
-			for(auto nameIt = enabledCollections.begin(); nameIt != enabledCollections.end(); nameIt++) {
-				if(*nameIt == selectedCollection) {
-					enabledCollectionName = selectedCollection;
-					break;
-				}
+	std::string selectedCollection = defaultScreenSaverCollection->getSelected();
```

Comment:
I am ok with this change in concept - it is what made sense - but this was here to handle a few edge cases, if I recall from my testing, where I could either get no element selected, or more than one element selected for some reason.
One was: the user has selected a collection (that's saved in settings) that has since been deleted (the file has been removed). This would have nothing selected at the time.
If the user deactivates the selected collection, I believe that this is also crashed, or if he went to the list and then selected another one. 
Maybe these have been mitigated by other changes in the code afterwards, but these were very deliberate from my end, as - ideally - I would have fixed it directly on OptionListComponent but I did not want to go down that rabbit hole right now.

Reply:
Multiple selections would be odd, as the collection list for the screensaver entry  is a single select list. getSelected() should be fine all the time.

The other case I can reproduce:
1. Have some custom collection set for the screensaver
2. Disable that collection in the custom collections
3. Exit dialog, reenter dialog (calling then getSelected() will fail due an assertion not matched)
4. Then exit dialog again (assertion fails)

This could be mitigated if iterated twice over the enabled custom collections on dialog creation. If no match in the first iteration do set and add <default> with selected == true.

Reply:
I'm not sure I follow the suggestion, but I believe you. Yes, that was one of the scenarios I got into. 

The thing is just that that component is so finicky that I'd rather play it safe - the logic currently there should also be a way to prevent this crash, or am I mistaken? 

Unless there's something super egregious with the current way the safeguards are implemented, I'd rather keep them as such - though I'd be happy to add comments there to explain why they are in place so that it's easy to understand why the approach was chosen.